massa-execution/src/vm.rs

```diff
@@ -174,6 +177,63 @@ impl VM {
         context.ledger_step.caused_changes.clone()
     }
 
+    /// Run code in read-only mode
+    pub(crate) fn run_read_only(
+        &self,
+        slot: Slot,
+        max_gas: u64,
+        simulated_gas_price: Amount,
+        bytecode: Vec<u8>,
+        address: Option<Address>,
+        result_sender: oneshot::Sender<ExecuteReadOnlyResponse>,
+    ) {
+        let saved_changes = {
+            let mut context = self.execution_context.lock().unwrap();
+
+            // Set the call stack, using the provided address, or a random one.
+            let address = address.unwrap_or_else(|| {
```

Comment:
the specification should be updated to reflect this random-address-by-default behavior

Reply:
done